MediaBrowser.Common/Providers/ProviderIdParsers.cs

```diff
@@ -0,0 +1,119 @@
+        /// <param name="imdbId">The parsed IMDb id.</param>
+        /// <returns>True if parsing was successful, false otherwise.</returns>
+        public static bool TryParseImdbId(string text, [NotNullWhen(true)] out string? imdbId)
+        {
+            var span = text.AsSpan();
+            var tt = "tt".AsSpan();
+
+            while (true)
+            {
+                var ttPos = span.IndexOf(tt);
+                if (ttPos == -1)
+                {
+                    imdbId = default;
+                    return false;
+                }
+
+                span = span.Slice(ttPos + tt.Length);
+
+                int i = 0;
+                // IMDb id has a maximum of 8 digits
+                int max = span.Length > 8 ? 8 : span.Length;
+                for (; i < max; i++)
+                {
+                    var c = span[i];
+
+                    if (c < '0' || c > '9')
+                    {
+                        break;
+                    }
+                }
+
+                // IMDb id has a minimum of 7 digits
+                if (i >= 7)
+                {
+                    imdbId = string.Concat(tt, span.Slice(0, i));
+                    return true;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Parses an TMDb id from a movie url.
+        /// </summary>
+        /// <param name="text">The text with the url to parse.</param>
+        /// <param name="tmdbId">The parsed TMDb id.</param>
+        /// <returns>True if parsing was successful, false otherwise.</returns>
+        public static bool TryParseTmdbMovieId(string text, [NotNullWhen(true)] out string? tmdbId)
+            => TryParseProviderId(text, "themoviedb.org/movie/", out tmdbId);
+
+        /// <summary>
+        /// Parses an TMDb id from a series url.
+        /// </summary>
+        /// <param name="text">The text with the url to parse.</param>
+        /// <param name="tmdbId">The parsed TMDb id.</param>
+        /// <returns>True if parsing was successful, false otherwise.</returns>
+        public static bool TryParseTmdbSeriesId(string text, [NotNullWhen(true)] out string? tmdbId)
+            => TryParseProviderId(text, "themoviedb.org/tv/", out tmdbId);
+
+        /// <summary>
+        /// Parses an TVDb id from a url.
+        /// </summary>
+        /// <param name="text">The text with the url to parse.</param>
+        /// <param name="tvdbId">The parsed TVDb id.</param>
+        /// <returns>True if parsing was successful, false otherwise.</returns>
+        public static bool TryParseTvdbId(string text, [NotNullWhen(true)] out string? tvdbId)
+            => TryParseProviderId(text, "thetvdb.com/?tab=series&id=", out tvdbId);
+
+        private static bool TryParseProviderId(string text, string searchString, [NotNullWhen(true)] out string? providerId)
+        {
+            var span = text.AsSpan();
+            var searchSpan = searchString.AsSpan();
+
+            while (true)
```

Comment:
Since this is a private function I don't think there's any need to loop when you're just looking for an url. It's either there or it isn't. I don't see it occurring multiple times organically.